src/blueprints/Do.Blueprints.Service.Application/ExceptionHandling/Default/GenericADOExceptionHandler.cs

```diff
@@ -0,0 +1,13 @@
+using NHibernate.Exceptions;
+
+namespace Do.ExceptionHandling.Default;
+
+public class GenericADOExceptionHandler : IExceptionHandler
+{
+    public bool CanHandle(Exception ex) => ex is GenericADOException;
```

Comment:
let's not handle every generic ado exception, only when its inner exception is invalid data